providers/fips/fipsprov.c

```diff
@@ -79,27 +77,37 @@ static OSSL_FUNC_BIO_vsnprintf_fn *c_BIO_vsnprintf;
 static OSSL_FUNC_self_test_cb_fn *c_stcbfn = NULL;
 static OSSL_FUNC_core_get_libctx_fn *c_get_libctx = NULL;
 
+typedef struct {
+    const char *option;
+    unsigned char enabled;
+    char non_default[2];
+} FIPS_OPTION;
+
 typedef struct fips_global_st {
     const OSSL_CORE_HANDLE *handle;
     SELF_TEST_POST_PARAMS selftest_params;
-    int fips_security_checks;
-    int fips_tls1_prf_ems_check;
-    const char *fips_security_check_option;
-    const char *fips_tls1_prf_ems_check_option;
+    FIPS_OPTION fips_security_checks;
+    FIPS_OPTION fips_tls1_prf_ems_check;
+    FIPS_OPTION fips_restricted_drgb_digests;
 } FIPS_GLOBAL;
 
+static ossl_inline void init_fips_option(FIPS_OPTION *opt, unsigned char enabled)
```

Comment:
unsigned char enabled seems like a strange thing for a boolean?

Reply:
It's better than signed char :smiley:  It will actually be passed as a unsigned int.